contrib/containers/ci/Dockerfile

```diff
@@ -26,13 +26,32 @@ RUN apt-get update && apt-get install $APT_ARGS \
     libtool \
     m4 \
     pkg-config \
-    python3 \
-    python3-dev \
-    python3-pip \
-    python3-setuptools \
     zlib1g-dev
 
-# Python stuff
+# Python setup
+# PYTHON_VERSION should match the value in .python-version
+ARG PYTHON_VERSION=3.8.16
+RUN apt-get update && apt-get install $APT_ARGS \
+    ca-certificates \
+    libbz2-dev \
+    libffi-dev \
+    liblzma-dev \
+    libncurses5-dev \
+    libncursesw5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    llvm \
+    make \
+    tk-dev \
+    xz-utils
+ENV PYENV_ROOT "/usr/local/pyenv"
```

Comment:
Because `buster` ships with Python 3.7 ([source](https://packages.debian.org/buster/python3)) and our build scripts are no longer compatible with it, the minimum version required in Python 3.8.